src/main/js/utils/Grade.js

```diff
@@ -0,0 +1,72 @@
+import queryString from 'query-string';
+import { gpaSearchUrl } from '../constants/resources';
+
+/**
+ * Returns the estimated gpa of a course
+ * @param course is the course to get the GPA of
+ * @param grades is the array containing courses with grade distribution data
+ */
+function getGPA(course, grades) {
+  let totalGPA = 0;
+  let number = 0;
+
+  const instructorFullName = course.instructor.split(' ');
+  const instructorLastName = instructorFullName[instructorFullName.length - 1];
+
+  grades.forEach((grade) => {
+    if (instructorLastName === 'Staff' && grade.subject === course.subject && grade.courseNumber === course.courseNumber) {
+      totalGPA += grade.gpa;
+      number += 1;
+    } else if (grade.subject === course.subject
+        && grade.courseNumber === course.courseNumber
+        && grade.instructor === instructorLastName) {
+      totalGPA += grade.gpa;
+      number += 1;
+    }
+  });
+
+  const gpa = totalGPA / number;
+  return !Number.isNaN(gpa) ? gpa.toFixed(2) : 0.00;
+}
+
+/**
+ * Returns a string that represents the query value when calling the GPA API
+ * @param courseList is a 2d array containing the list of courses
+ */
+function getQuerySearch(courseList) {
+  const courseArr = [];
+  courseList.forEach(subArray => courseArr.push(`${subArray[0].subject}${subArray[0].courseNumber}`));
+
+  return queryString.stringify({ query: courseArr });
+}
+
+/**
+ * Adds a gpa field to each course inside the courselist
+ * @param courseList is a 2d array containing the list of courses
+ */
+function addGPA(courseList) {
+  const query = getQuerySearch(courseList);
+  const site = `${gpaSearchUrl}?${query}`;
+
+  return fetch(site)
+    .then(res => res.json()).then(res => res._embedded).then(res => res.gpa) // eslint-disable-line
+    .then((res) => {
+      const finalList = [];
+      courseList.forEach((subArray, currentIndex) => {
+        const gpaArray = subArray.map((course) => { // eslint-disable-line no-param-reassign
```

Comment:
Thinking about it more, I don't think it's a good practice to just add a gpa field to Schedule like this. I think we should instead create a map of courses like below and pass that as a parameter/store it in the state to use wherever we need it. Doing it this way means the GPA searching can be used for multiple features, not just the generator. Thoughts?
```
{
   CS2114: {
      Ellis: [
          <CourseGPAs for Ellis>
      ],
      ...
   },
   ...
}
```

Reply:
Done. Chose the bottom option

Reply:
That snippet of code adds a 'gpa' attribute to a COURSE object, not schedule btw.

Reply:
but I agree that a state to keep track of GPA courses should be held in a redux store.

Reply:
Fixed.